src/sentry/seer/math.py

```diff
@@ -16,6 +16,45 @@ def laplace_smooth(probabilities: list[float], alpha: float = 1e-3) -> list[floa
     return [(value + alpha) / (total + alpha * len(probabilities)) for value in probabilities]
 
 
+def entropy(xs: list[float]) -> float:
+    """
+    Shannon entropy. Quantifies the amount of uncertainty in a probability distribution.
+
+    Equation:
+        H = -sum(p(x) * log2(p(x)))
+
+    Parameters:
+        probabilities: A list of non-negative floating point values which sum to 1.
+    """
+
+    # This is the scipy implementation. Copied from here:
+    # https://github.com/scipy/scipy/blob/ce4b43097356dfc42504d81d6164b73ee0896c71/scipy/special/_convex_analysis.pxd#L8-L16
```

Comment:
Do we need to add a link to their license too?
Or copy it in some form to this file (e.g: header?)

at the same sha:
https://github.com/scipy/scipy/blob/ce4b43097356dfc42504d81d6164b73ee0896c71/LICENSE.txt

Reply:
I am definitely not qualified to answer that question 😂.  Should we ask legal?  We have scipy installed in Seer would that require the same attribution?

Reply:
Its BSD licensed so apparently free to distribute how we wish? https://appmaster.io/blog/what-is-the-bsd-license